### PR TITLE
[Perf][Qwen3-TTS] Skip per-step hidden-state D2H via hidden pooler payload opt-out

### PR DESCRIPTION
**Pull request by linyueqian**
## Purpose

W1 of #4855. Qwen3-TTS stage 1 (code2wav) consumes runtime audio codes, not hidden states, but the runner still materializes a synchronous CPU copy of the full hidden states **every decode step** for the inter-stage pooler payload:

```python
# vllm_omni/worker/gpu_ar_model_runner.py
if hs_for_cache is None:
    hidden_states_cpu = hidden_states[:num_tokens_unpadded].detach().to("cpu").contiguous()
```

py-spy at c=64 attributes 24% of stage-0 on-CPU samples to this line. The opt-out flag and its payload-builder gates already exist (`omni_pooler_payload_include_hidden`, set by qwen3_omni, honored by `_model_omni_pooler_payload_include_hidden()` in the output builder); qwen3_tts just never set it, and this one materialization site did not check it.

This PR (2 files, ~8 lines): the talker declares `omni_pooler_payload_include_hidden = False`, and the copy site gates on the existing flag. Models that do not opt out keep the default `include_hidden=True` path byte-for-byte unchanged.

## Test Plan

A/B on 1x H20, Qwen3-TTS-12Hz-1.7B-CustomVoice, c=64, `max_num_seqs=64`, seed-tts-text en, 300 requests per arm, both arms completing 300/300 with audio (streaming and non-streaming verified):

| | throughput (audio-s/s) | mean TTFP | p99 TTFP | stage-0 host CPU | copy in py-spy profile |
|---|---|---|---|---|---|
| main | 45.7 | 1056 ms | 3270 ms | 59% of a core | 24.1% of samples |
| this PR | 51.7 (+13%) | 479 ms (-55%) | 1067 ms (-67%) | 24% of a core | gone |

Unit suites, run on the PR head (`ef13bdae`, includes the merge of main) on H20 with vLLM 0.24:

```
pytest -q tests/entrypoints/openai_api/test_serving_speech.py tests/entrypoints/openai_api/test_tts_adapter.py
202 passed, 19 warnings in 4.83s
```

**vLLM Version:** 0.24.0

### CODE REVIEW

**Comment by Sy0307** [comment]
Plz fix DCO and these changes look nice to me :)

**Review by Sy0307 — APPROVED** [review]

**Comment by hsliuustc0106** [comment]
does this happen to other models as well?

**Comment by linyueqian** [comment]
Good question. This specific sync copy only fires for models that opt out of full prefix-cached hidden states (`requires_full_prefix_cached_hidden_states = False`) while prefix caching is enabled, and on main that is exactly two models: qwen3_tts (`qwen3_tts_talker.py:311`) and higgs_audio_v3 (`higgs_audio_v3_talker.py:236`).

Higgs v3 is in the same situation today: its deploy (`higgs_multimodal_qwen3.yaml`) also ships `enable_prefix_caching: true`, and its stage 1 consumes runtime audio codes rather than hidden states, so the same one-line opt-out should apply. I kept this PR to qwen3_tts because that is the config I actually benchmarked; I can add the higgs flag in a tiny follow-up once I have run the same A/B there.

All other TTS models keep `requires_full_prefix_cached_hidden_states = True`, so with prefix caching on they take the async prefix-cache write path instead of this synchronous branch, and without prefix caching the copy never runs. They are not affected either way.

**Comment by linyueqian** [comment]
@Sy0307 DCO is fixed (branch squashed to a single signed commit; the check passes now).

On the main Buildkite lane: builds 12052 and 12055 both failed on this head, but I cannot see the failing step name without Buildkite org access, and everything I can run or see is green:

- `tests/entrypoints/openai_api/test_serving_speech.py` + `test_tts_adapter.py` on the exact head (`ef13bdae`): 202 passed
- `tests/worker/`: 113 passed
- Intel CI, NPU CI, GHA builds, pre-commit, docs: all passing
- main itself passes (build 12050), so the pipeline is not globally red

Could someone with Buildkite access share which step is failing (or whether it is a known-flaky cell)? If it is genuinely this change I will fix it; the change is a two-line gate scoped to models that set `omni_pooler_payload_include_hidden = False` (only qwen3_tts here), and the c=64 e2e on H20 ran 300/300 with audio in both streaming and non-streaming modes.

**Comment by linyueqian** [comment]
Update: the CI failure is real and caused by this change. Converting to draft while I rework it.

The failing step is `tests/e2e/online_serving/test_qwen3_tts_base.py::test_text_to_audio_001[async_chunk]` (Base task + voice clone). Reproduced on H20 at this PR's head with a single cold Base+ref_audio request, A/B on the flag alone:

| arm | single Base voice-clone request |
|---|---|
| opt-out ON (this PR) | 200 OK, 29.5 s |
| opt-out OFF | 200 OK, 8.7 s |

So the request completes but runs ~3.4x slower on the Base/codec-streaming path; on CI's L4 with several concurrent requests that exceeds `DEFAULT_AUDIO_SPEECH_TIMEOUT_S`, hence the APITimeoutError. My earlier validation only exercised CustomVoice (which improves, +13% tput at c=64), so I missed this.

Root-cause direction: for opt-out prefix-cache models the scheduled hidden payload is load-bearing on the Base path (the async-output/scheduled-hidden gating in `gpu_ar_model_runner.py` around `:1554-1563` keys off `_model_omni_pooler_payload_include_hidden()`), so a model-level opt-out is too blunt. I will rework this so the hidden payload is skipped only where nothing consumes it (per-request or consumer-derived), re-run the A/B on both Base voice-clone and CustomVoice, and un-draft once both are green.

**Comment by linyueqian** [comment]
Correction: my previous comment was wrong. This PR does not slow the Base path. The "3.4x regression" in my repro was a cold-vs-warm `torch.compile` cache confound in my own A/B, and I am retracting the claim with the same prominence I made it.

What the controlled experiment shows (per-arm isolated caches via `VLLM_CACHE_ROOT`/`TORCHINDUCTOR_CACHE_DIR`/`TRITON_CACHE_DIR`, arm order swapped):

| arm | single cold Base voice-clone request |
|---|---|
| opt-out OFF (run first this time) | 200 OK, 29.3 s |
| opt-out ON (this PR) | 200 OK, 29.6 s |

Parity within noise. My earlier "8.7 s vs 29.5 s" compared a warm-cache second run against a cold-cache first run: during the first arm's request window ~1,000 Inductor artifacts were being written by `Qwen3CodePredictor._setup_compile()` (lazy `torch.compile` + CUDA-graph capture inside the first request); the second arm rode that warm cache. Per-step decode actually favors this PR (TPOT 10.45 ms vs 11.18 ms), outputs are byte-identical, and a code audit found no consumer of the hidden payload anywhere on the qwen3-tts Base path (stage 1 consumes `codes.*` only; the talker's own loop uses GPU-resident buffers).

The real issue this uncovered is orthogonal to this PR: the code predictor's ~25 s lazy compile runs inside the first request instead of at engine warmup, which on a slow CI L4 can push the first e2e request toward `DEFAULT_AUDIO_SPEECH_TIMEOUT_S` regardless of this change. I will propose eager code-predictor warmup at startup as a separate small PR.

Un-drafting and re-adding the CI labels for a fresh verdict.

**Comment by linyueqian** [comment]
The failing step (`test_qwen3_tts_base.py::test_text_to_audio_001[async_chunk]`) is #4883 — it fails identically on main at `0798b0bf` (build 12043), which is this PR's base commit. Not introduced by this change; see the diagnostic I posted on #4883.

**Comment by amy-why-3459** [comment]
For Qwen3-TTS, prefix caching adds per-step cache bookkeeping with little benefit on streaming decode, and it also blocks the async omni output path. This change disables prefix caching in the deploy config and enables three talker flags: use_async_omni_output = True, eager_omni_postprocess_before_async_output = True, and omni_pooler_payload_include_hidden = False. Together they run postprocess() on live GPU tensors, asynchronously copy only the needed multimodal payloads (e.g. codes.audio) to CPU, and skip hidden-state D2H in the inter-stage snapshot. Under 8 concurrent requests, audio RTF  0.54 -> 0.44. 
```
+        self.use_async_omni_output = True
+        self.eager_omni_postprocess_before_async_output = True
+        self.omni_pooler_payload_include_hidden = False
```
```
-    enable_prefix_caching: true
+    enable_prefix_caching: false
```
